### PR TITLE
[SYCL][UR] Bump UR version and add missing values to pi2ur

### PR DESCRIPTION
**Pull request by callumfare**
Includes a large number of missing `pi_result` mappings, so previously reported errors should no longer map to just `PI_ERROR_UNKNOWN`. NFCI for the adapters.

### CODE REVIEW

File: sycl/plugins/unified_runtime/ur/adapters/cuda/kernel.cpp [hunk 1]
```diff
@@ -186,9 +186,11 @@ UR_APIEXPORT ur_result_t UR_APICALL urKernelGetNativeHandle(
...
 }
 
-UR_APIEXPORT ur_result_t UR_APICALL
-urKernelSetArgValue(ur_kernel_handle_t hKernel, uint32_t argIndex,
-                    size_t argSize, const void *pArgValue) {
+UR_APIEXPORT ur_result_t UR_APICALL urKernelSetArgValue(
+    ur_kernel_handle_t hKernel, uint32_t argIndex, size_t argSize,
+    const ur_kernel_arg_value_properties_t *pProperties,
+    const void *pArgValue) {
+  std::ignore = pProperties;
```
Comment (jchlanda) [inline sycl/plugins/unified_runtime/ur/adapters/cuda/kernel.cpp:193]: Nice :)

**Review by jchlanda — APPROVED** [review]

**Comment by jandres742** [comment]
Failure on Gen9 is with OpenCL, not L0. @callumfare patch doesn't touch anything related to OpenCL, so this seems something flaky.

```
$ ":" "RUN: at line 6"
note: command had no output on stdout or stderr
$ "env" "ONEAPI_DEVICE_SELECTOR=ext_oneapi_level_zero:gpu" "/__w/llvm/llvm/build-e2e/Regression/commandlist/Output/gpu.cpp.tmp.out"
# command output:
NThreads: GPU 8
   shared Queue: true
   array size: 1048576 (1<<20)
   SyCL items: 1048576
   iter GPU:   200
selected dev: Intel(R) UHD Graphics


                           subm   strt    end    run
GPU[1]:  nPrimes: 82024    0.00   0.20   2.87   2.66      0.00   2.88   2.88
GPU[6]:  nPrimes: 82024    2.88   2.87   5.52   2.65      0.01   5.55   5.54
GPU[5]:  nPrimes: 82024    5.55   5.52   8.17   2.65      0.01   8.21   8.19
GPU[2]:  nPrimes: 82024    8.21   8.17  10.81   2.64      0.01  10.86  10.85
GPU[4]:  nPrimes: 82024   10.86  10.82  13.47   2.65      0.01  13.52  13.51
GPU[3]:  nPrimes: 82024   13.52  13.47  16.37   2.90      0.01  16.43  16.42
GPU[7]:  nPrimes: 82024   16.43  16.37  19.07   2.71      0.01  19.15  19.14
GPU[0]:  nPrimes: 82024   19.15  19.08  21.76   2.69      0.02  21.85  21.84
GPU average time: 2.69
total time: 21.87
Passed

$ ":" "RUN: at line 6"
note: command had no output on stdout or stderr
$ "env" "ONEAPI_DEVICE_SELECTOR=opencl:gpu" "/__w/llvm/llvm/build-e2e/Regression/commandlist/Output/gpu.cpp.tmp.out"
# command output:
NThreads: GPU 8
   shared Queue: true
   array size: 1048576 (1<<20)
   SyCL items: 1048576
   iter GPU:   200
selected dev: Intel(R) UHD Graphics


error: command failed with exit status: -11
```

**Comment by jandres742** [comment]
@intel/llvm-gatekeepers  : do you want to redeploy to see if the failures are indeed flaky or can this be merged now?

**Comment by jandres742** [comment]
@intel/llvm-gatekeepers : could we merge this or could you suggest next steps?

**Pull request merged**